pkg/executor/wasm/executor.go

```diff
@@ -80,9 +91,13 @@ func (e *Executor) Start(ctx context.Context, request *executor.RunCommandReques
 	// limit is not specified as a multiple of that.
 	engineConfig := wazero.NewRuntimeConfig().WithCloseOnContextDone(true)
 	if request.Resources.Memory > 0 {
-		const pageSize = 65536
-		pageLimit := request.Resources.Memory/pageSize + math.Min(request.Resources.Memory%pageSize, 1)
-		engineConfig = engineConfig.WithMemoryLimitPages(uint32(pageLimit))
+		requestedPages := request.Resources.Memory/WASM_PAGE_SIZE + math.Min(request.Resources.Memory%WASM_PAGE_SIZE, 1)
+		if requestedPages > WASM_MAX_PAGES_LIMIT {
+			err := fmt.Errorf("requested memory exceeds maximum limit: %d > %d", requestedPages, WASM_MAX_PAGES_LIMIT)
```

Comment:
@rossjones the error being returned is correct, but hard to understand for an average error. Should we consider returning error like "Requested memory exceeds the wasm limit - %d > 4GB"

Reply:
Yes, that would be a better error message.

Reply:
ok Some BDD tests, cool!

Reply:
@rossjones i have updated the error msg. But just to be clear do u have a specific wasm program in mind?

Reply:
I meant for the tests. And how should i add it, can we afford git-lfs?

Reply:
For now, i can write a unit test by leveraging `NewExecutor()`